runtime/src/axum/mod.rs

```diff
@@ -140,67 +152,68 @@ struct RouterBuilder {
 }
 
 impl RouterBuilder {
-    pub fn new() -> Self {
+    fn new() -> anyhow::Result<Self> {
         let engine = Engine::default();
 
         let mut linker: Linker<WasiCtx> = Linker::new(&engine);
-        wasmtime_wasi::add_to_linker(&mut linker, |s| s).unwrap();
+        wasmtime_wasi::add_to_linker(&mut linker, |s| s)?;
 
-        Self {
+        Ok(Self {
             engine,
             linker,
             src: None,
-        }
+        })
     }
 
-    pub fn src<P: AsRef<Path>>(mut self, src: P) -> Self {
+    fn src<P: AsRef<Path>>(mut self, src: P) -> Self {
         self.src = Some(src.as_ref().to_path_buf());
         self
     }
 
-    pub fn build(self) -> Router {
-        let file = self.src.unwrap();
-        let module = Module::from_file(&self.engine, file).unwrap();
+    fn build(self) -> anyhow::Result<Router> {
+        let file = self.src.expect("module path should be set");
+        let module = Module::from_file(&self.engine, file)?;
 
         for export in module.exports() {
             println!("export: {}", export.name());
         }
-        let inner = RouterInner {
+
+        Ok(Router {
             linker: self.linker,
             engine: self.engine,
             module,
-        };
-        Router { inner }
+        })
     }
 }
 
 #[derive(Clone)]
-struct RouterInner {
+struct Router {
     linker: Linker<WasiCtx>,
     engine: Engine,
     module: Module,
 }
 
-impl RouterInner {
+impl Router {
     /// Send a HTTP request with body to given endpoint on the axum-wasm router and return the response
-    pub async fn handle_request(
+    async fn handle_request(
         &mut self,
         req: hyper::Request<Body>,
-    ) -> Result<Response<Body>, Infallible> {
+    ) -> anyhow::Result<Response<Body>> {
         let wasi = WasiCtxBuilder::new()
             .inherit_stdio()
             .inherit_args()
-            .unwrap()
+            .context("failed to read args")?
             .build();
 
         let mut store = Store::new(&self.engine, wasi);
-        self.linker
-            .module(&mut store, "axum", &self.module)
-            .unwrap();
+        self.linker.module(&mut store, "axum", &self.module)?;
 
-        let (mut parts_stream, parts_client) = UnixStream::pair().unwrap();
-        let (mut body_write_stream, body_write_client) = UnixStream::pair().unwrap();
-        let (body_read_stream, body_read_client) = UnixStream::pair().unwrap();
+        let (mut parts_stream, parts_client) =
+            UnixStream::pair().context("failed to open unixstream")?;
+        let (mut body_write_stream, body_write_client) =
+            UnixStream::pair().context("failed to open unixstream")?;
+        let (body_read_stream, body_read_client) =
+            UnixStream::pair().context("failed to open unixstream")?;
```

Comment:
Maybe we should rather have "failed to open unix stream for parts" else we won't know which one failed

Reply:
Good point.